[BUG ] Intermittent error when applying batches

Sometimes batch apply fails with:
```
Caused by: java.util.NoSuchElementException: next on empty iterator
	at scala.collection.Iterator$$anon$19.next(Iterator.scala:973)
	at scala.collection.Iterator$$anon$19.next(Iterator.scala:971)
	at scala.collection.IterableOps.head(Iterable.scala:222)
	at scala.collection.IterableOps.head$(Iterable.scala:222)
	at scala.collection.AbstractIterable.head(Iterable.scala:935)
	at com.sneaksanddata.arcane.microsoft_synapse_link.services.streaming.consumers.IndexedStagedBatchesImpl.getOptimizationRequest(IndexedStagedBatchesImpl.scala:30)
	at com.sneaksanddata.arcane.framework.services.streaming.base.StagedBatchProcessor.runMaintenanceTasks$$anonfun$1(StagedBatchProcessor.scala:94)
	at com.sneaksanddata.arcane.framework.services.streaming.base.StagedBatchProcessor.runMaintenance$$anonfun$1(StagedBatchProcessor.scala:105)
	at scala.Option.map(Option.scala:242)
	at com.sneaksanddata.arcane.framework.services.streaming.base.StagedBatchProcessor.runMaintenance(StagedBatchProcessor.scala:105)
	at com.sneaksanddata.arcane.framework.services.streaming.base.StagedBatchProcessor.runMaintenanceTasks(StagedBatchProcessor.scala:94)
	at com.sneaksanddata.arcane.framework.services.streaming.base.StagedBatchProcessor.runMaintenanceTasks$(StagedBatchProcessor.scala:56)
	at com.sneaksanddata.arcane.framework.services.streaming.processors.batch_processors.MergeBatchProcessor.runMaintenanceTasks(MergeBatchProcessor.scala:16)
	at com.sneaksanddata.arcane.framework.services.streaming.processors.batch_processors.MergeBatchProcessor.process$$anonfun$1$$anonfun$1$$anonfun$2$$anonfun$2(MergeBatchProcessor.scala:29)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1077)
```
~~By the looks of it, we never execute table optimize~~. We do, just saw an example. So not sure what this one is related to.

--- a/src/main/scala/services/streaming/consumers/IcebergSynapseConsumer.scala
+++ b/src/main/scala/services/streaming/consumers/IcebergSynapseConsumer.scala
@@ -35,7 +35,7 @@ class IcebergSynapseConsumer(stagingProcessor: StagingProcessor,
    * @return ZSink (stream sink for the stream graph).
    */
   override def consume: ZSink[Any, Throwable, IncomingBatch, Any, Unit] =
-    stagingProcessor.process(toInFlightBatch) >>> mergeProcessor.process >>> disposeBatchProcessor.process >>> lifetimeGuard >>> logResults
+    stagingProcessor.process(toInFlightBatch).filter(_.groupedBySchema.nonEmpty)  >>> mergeProcessor.process >>> disposeBatchProcessor.process >>> lifetimeGuard >>> logResults
 
   private def lifetimeGuard: ZPipeline[Any, Throwable, MergeBatchProcessor#BatchType, MergeBatchProcessor#BatchType] =
     ZPipeline.takeUntil(_ => streamLifetimeService.cancelled)

